src/geometries/WireframeGeometry.js

```diff
@@ -11,106 +13,105 @@ function WireframeGeometry( geometry ) {
 
 	this.type = 'WireframeGeometry';
 
-	var edge = [ 0, 0 ], hash = {};
+	// buffer
 
-	var keys = [ 'a', 'b', 'c' ];
+	var vertices = [];
+
+	// helper variables
+
+	var i, j, l, o, ol;
+	var edge = [ 0, 0 ], edges = {}, e;
+	var key, keys = [ 'a', 'b', 'c' ];
+	var vertex;
+
+	// different logic for Geometry and BufferGeometry
 
 	if ( geometry && geometry.isGeometry ) {
 
-		var vertices = geometry.vertices;
-		var faces = geometry.faces;
-		var numEdges = 0;
+		// create a data structure that contains all edges without duplicates
 
-		// allocate maximal size
-		var edges = new Uint32Array( 6 * faces.length );
+		var faces = geometry.faces;
 
-		for ( var i = 0, l = faces.length; i < l; i ++ ) {
+		for ( i = 0, l = faces.length; i < l; i ++ ) {
```

Comment:
I try to avoid all redefinitions in our code. All declaration of variables via `var` are actually hoisted to the top of the function. So the scope is not block but `function`, even if we declare the variable inside a loop of `if`-statement. 

Therefore i declare all variables at the top, that are multiple used in the implementation. Variables that a used in just one single block are declared inside of it.

Douglas Crockford argues like [this](https://plus.google.com/u/0/+DouglasCrockfordEsq/posts/2JY9bjaup1g):

_JSLint now warns when a var is defined that has the same name as something in the outer scope. This is confusing because the reader cannot easily tell which variable he is looking at. It is sometimes an error because the new variable is accidentally hiding the old one. In some cases, the old one is the one intended._

We can avoid this problem if we switch some day to `let` and `const`.